### PR TITLE
integrate evm accounts and evm

### PR DESCRIPTION
**Pull request by xlc**
More cleanup needed later, but get it working first

### CODE REVIEW

**Review by zjb0807 — APPROVED** [review]

File: modules/evm/src/runner/native.rs [hunk 1]
```diff
@@ -107,9 +119,16 @@ impl<T: Trait> RunnerT<T> for Runner<T> {
...
+			debug::debug!(
+				target: "evm",
+				"create-result: create_info {:?}",
+				create_info
+			);
+
 			match reason {
 				ExitReason::Succeed(_) => match substate.gasometer.record_deposit(out.len()) {
 					Ok(()) => {
+						substate.inc_nonce(address);
```
Comment (ermalkaleci) [inline modules/evm/src/runner/native.rs:131]: @xlc why incrementing contract nonce?

File: modules/evm/src/runner/native.rs [hunk 2]
```diff
@@ -132,32 +151,33 @@ impl<T: Trait> RunnerT<T> for Runner<T> {
...
+			salt,
+		});
 
-			let code_hash = H256::from_slice(Keccak256::digest(&init).as_slice());
-			let address = substate.create_address(CreateScheme::Create2 {
-				caller: source,
-				code_hash,
-				salt,
-			});
+		substate.inc_nonce(address);
```
Comment (ermalkaleci) [inline modules/evm/src/runner/native.rs:178]:
```suggestion
		substate.inc_nonce(source);
```

File: modules/evm-accounts/src/lib.rs [hunk 3]
```diff
@@ -260,7 +278,7 @@ where
 	T::AccountId: From<AccountId32>,
 {
 	fn into_h160(account_id: AccountId32) -> H160 {
-		EvmAddresses::<T>::get(&Into::<T::AccountId>::into(account_id))
+		EvmAddresses::<T>::get(&Into::<T::AccountId>::into(account_id)).unwrap_or_default()
```
Comment (ermalkaleci) [inline modules/evm-accounts/src/lib.rs:281]: @xlc is it safe to return empty H160?

File: modules/evm/src/runner/native.rs (hunk 1 above)
Comment (xlc) [reply modules/evm/src/runner/native.rs:131]: To confirm eth 161 https://github.com/ethereum/EIPs/blob/master/EIPS/eip-161.md

File: modules/evm-accounts/src/lib.rs (hunk 3 above)
Comment (xlc) [reply modules/evm-accounts/src/lib.rs:281]: This was the old behavior

https://github.com/AcalaNetwork/Acala/issues/525 to cleanup this

File: modules/evm/src/runner/native.rs (hunk 2 above)
Comment (xlc) [reply modules/evm/src/runner/native.rs:178]: this is intended per eip 161

Comment (ermalkaleci) [reply modules/evm/src/runner/native.rs:178]: that's called here https://github.com/AcalaNetwork/Acala/pull/521/files/e0300549204ad9189127eb79c3145f5342bae0db#diff-9a78924138ebd2942f83151d282cf9216b5e8986eedd9a0051d79baa11b64409R202
I don't see caller nonce to be incremented

Comment (ermalkaleci) [reply modules/evm/src/runner/native.rs:178]: will fix on #528

**Pull request merged**